### ISSUE TITLE
UUID error when starting app in watch mode

### ISSUE THREAD

**Issue by ChrisDusyk** [nestjs/nest#4833]
## Current behavior
Just today I cloned my app to a new computer, ran `yarn install` and tried to run it in dev mode. However I'm now getting this error:

```
[10:43:47 a.m.] Starting compilation in watch mode...

[10:43:49 a.m.] Found 0 errors. Watching for file changes.

internal/modules/cjs/loader.js:625
  throw e;
  ^

Error: No valid exports main found for 'C:\Personal Code\Cribbage\cribbage-server\node_modules\@nestjs\common\node_modules\uuid'
    at resolveExportsTarget (internal/modules/cjs/loader.js:622:9)
    at applyExports (internal/modules/cjs/loader.js:499:14)
    at resolveExports (internal/modules/cjs/loader.js:548:12)
    at Function.Module._findPath (internal/modules/cjs/loader.js:654:22)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:953:27)
    at Function.Module._load (internal/modules/cjs/loader.js:859:27)
    at Module.require (internal/modules/cjs/loader.js:1028:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (C:\Personal Code\Cribbage\cribbage-server\node_modules\@nestjs\common\decorators\core\injectable.decorator.js:3:16)
    at Module._compile (internal/modules/cjs/loader.js:1139:30) {
  code: 'MODULE_NOT_FOUND'
}
```

## Input Code

<https://github.com/ChrisDusyk/nest-error-example>

## Expected behavior

Nestjs starts in watch mode.

## Environment

<pre><code>
Nest version: 7.1.5

 
For Tooling issues:
- Node version: 13.5.0
- Platform:  Windows 10

Others:

</code></pre>

**Comment by jmcdo29** [comment]
Duplicate of #4824

**Comment by tirathsharma098** [comment]
I faced same issue. 
I was using Typeorm (having uuid v11 by default), @nestjs/typeorm (having same v11 for uuid) already in the app.
When I needed uuid package I installed it v13
My Node v20 
When running app got the same error.

not resolved by v8 therefore again at latest, ror [ERR_REQUIRE_ESM]: require() of ES Module /home/learning/main/MyProjects/meanblogs/code-digit-server/node_modules/.pnpm/uuid@13.0.0/node_modules/uuid/dist-node/index.js from /home/learning/main/MyProjects/meanblogs/code-digit-server/dist/upload/upload.service.js not supported. Instead change the require of index.js in /home/learning/main/MyProjects/meanblogs/code-digit-server/dist/upload/upload.service.js to a dynamic import() which is available in all CommonJS modules. at Object.<anonymous> (/home/learning/main/MyProjects/meanblogs/code-digit-server/dist/upload/upload.service.js:58:16) { code: 'ERR_REQUIRE_ESM' } Node.js v20.18.2

Than today in AI era each time AI suggested me for 4 hours either degrade v to 8 for uuid or telling me my configuration is wrong or I am not using module instead using commonjs.

But my config was correct as app was running perfect before installing uuid.

After a lot of debug I checked versions for all and found that already existing library using v11 of uuid. 
As I removed v13 and installed v11 my app worked perfect again.

Hope it will help others.
